spring-cloud-starter-stream-common-tensorflow/pom.xml

```diff
@@ -21,6 +21,14 @@
             <groupId>org.tensorflow</groupId>
             <artifactId>proto</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.tensorflow</groupId>
+            <artifactId>libtensorflow</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.tensorflow</groupId>
+            <artifactId>libtensorflow_jni_gpu</artifactId>
```

Comment:
Would you mind to elaborate more here, please?

Looks like using this here we don't have choice in the target application: https://www.tensorflow.org/versions/master/install/install_linux#determine_which_tensorflow_to_install

I wonder what is going to happen if I don't have the proper GPU or I am just on Windows (although my Nvidia GPU is pretty good :smile: )

I mean shouldn't we leave such a choice for end-user?
Or doesn't it just have any effect if we don't meet GPU requirements and it just fallback to CPU mode?

Thanks

Reply:
Good point!
According to this doc: https://www.tensorflow.org/install/install_java 
> GPU acceleration is available via Maven only for Linux and only if your system meets the requirements for GPU.
So my assumption is that those dependencies will only be activated if the above conditions hold. For example it works find on my MacBook laptop using only the CPU there (e.g. the GPU dependencies are ignored).
Also with those additional maven dependencies i hoped we could address issues like this: https://stackoverflow.com/questions/46430512/tensorflow-train-a-model-with-gpus-and-inference-with-java-api

But having said this i acknowledge that i have not tested this functionality, so i will comment out those dependencies and open a separate issue for tackling this.